src/test/scala/chiselTests/naming/SuggestNameSpec.scala

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.naming
+
+import chisel3._
+import chisel3.aop.Select
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage}
+import chisel3.experimental.{dump, noPrefix, prefix, treedump}
+import chiselTests.{ChiselPropSpec, Utils}
+import chisel3.stage.{DesignAnnotation, NoRunFirrtlCompilerAnnotation}
+import firrtl.options.{Dependency, Phase, PhaseManager}
+import firrtl.options.phases.DeletedWrapper
+
+class SuggestNameSpec extends ChiselPropSpec with Utils {
+  implicit val minimumMajorVersion: Int = 12
+
+  property("0. Calling suggestName 2x should be a runtime deprecation") {
+    class Test extends Module {
+      {
+        val wire = {
+          val x = WireInit(0.U(3.W)).suggestName("mywire")
+          dontTouch(x)
+        }
+        wire.suggestName("somethingElse")
+      }
+    }
+    val (log, _) = grabLog(ChiselStage.emitVerilog(new Test()))
+    log should include(
+      "Calling suggestName (somethingElse, when already called with Some(mywire)) will become an error in Chisel 3.6"
+    )
+  }
+
+  property("1. Calling suggestName outside of a Builder context should be an error") {
+    class Test extends Module {
+      val wire = {
+        val x = WireInit(0.U(3.W))
+        dontTouch(x)
+      }
+    }
+
+    val pm = new PhaseManager(Seq(Dependency[chisel3.stage.phases.Checks], Dependency[chisel3.stage.phases.Elaborate]))
+    val test = pm
+      .transform(Seq(ChiselGeneratorAnnotation(() => new Test()), NoRunFirrtlCompilerAnnotation))
+      .collectFirst {
+        case d: DesignAnnotation[_] => d
+      }
+      .get
+      .design
```

Comment:
FYI there's a much simpler way to express this sort of thing:
```suggestion
var test: Test = null
ChiselStage.elaborate {
  test = new Test
  test
}
```
It's nasty and horrible but pithy 🙂. Scala 3 has a [neat feature](https://dotty.epfl.ch/docs/reference/experimental/cc.html) to make such hacks a compile-time error, but for the foreseeable future we can do this.

That being said I'm suggesting changing this to a warning for this PR anyway.

Reply:
😱